client/src/components/ConfigTemplates/SourceOptionCard.vue

```diff
@@ -0,0 +1,139 @@
+import { type IconDefinition } from "font-awesome-6";
+
+import type { FileSourceTemplateSummary } from "@/api/fileSources";
+import type { ObjectStoreTemplateSummary } from "@/api/objectStores.templates";
+import { markup } from "@/components/ObjectStore/configurationMarkdown";
+
+import Heading from "@/components/Common/Heading.vue";
+import TextSummary from "@/components/Common/TextSummary.vue";
+
+type OptionType = {
+    icon: IconDefinition;
+    title: string;
+};
+
+interface Props {
+    /** The source option to display */
+    sourceOption: FileSourceTemplateSummary | ObjectStoreTemplateSummary;
+    /** The route to navigate to when the user selects this option */
+    selectRoute: string;
+    /** The type of the source option */
+    optionType: OptionType;
+    /** Whether to display the card in a grid view */
+    gridView?: boolean;
+}
+
+const props = defineProps<Props>();
+
+function markdownToHTML(text: string) {
+    return markup(text ?? "", true);
+}
+</script>
+
+<template>
+    <div :key="props.sourceOption.id" class="source-option-card" :class="{ 'grid-view': props.gridView }">
+        <div class="source-option-card-content">
+            <div>
+                <div class="source-option-card-header">
+                    <Heading bold size="md" class="source-option-card-header-name">
+                        {{ props.sourceOption.name }}
+
+                        <BButton
+                            v-b-tooltip.hover.noninteractive
+                            variant="outline-primary"
+                            size="lg"
+                            :title="props.optionType?.title"
+                            class="inline-icon-button">
+                            <FontAwesomeIcon :icon="props.optionType?.icon" />
+                        </BButton>
+                    </Heading>
+
+                    <slot name="badges" class="source-option-card-header-type" />
+                </div>
+
+                <TextSummary
+                    v-if="props.sourceOption.description"
+                    class="source-option-card-description"
+                    is-html
+                    show-expand-text
+                    :description="markdownToHTML(props.sourceOption.description) ?? ''"
+                    :max-length="props.gridView ? 250 : 350" />
+            </div>
+
+            <BButton
+                :id="`select-button-${props.sourceOption.id}`"
+                v-b-tooltip.hover.noninteractive
+                variant="outline-primary"
+                class="source-option-card-select-button"
+                :class="{ 'source-option-card-select-button-grid': props.gridView }"
+                title="Select this option to create a new instance"
+                :to="props.selectRoute">
+                Select
+            </BButton>
+        </div>
+    </div>
+</template>
+
+<style scoped lang="scss">
+@import "theme/blue.scss";
+@import "_breakpoints.scss";
+
+.source-option-card {
```

Comment:
It seems like we are adding another separate card/grid display including styles here, like previously in panels and grids. We should consolidate these and use a shared component if we can.

Reply:
@guerler Thank you for pointing this out. You're right, but this would require a large clean-up (as we have several components using this UI) and code refactoring. I'll create the following PR(s) (before adding new cards and more duplications) to replace all existing cards with generic ones and reduce the redundancy. What do you think?

Reply:
We had the same discussion with regard to workflow and notification cards. I don't want to hold this PR up tho. Please follow-up with consolidating those, so cards look consistent and can be updated/fixed without having to change many locations manually.